### PR TITLE
Michael/hovering menu

### PR DESCRIPTION
**Pull request by msenejoa**

### CODE REVIEW

File: packages/databyss-documentation/website/package.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
```
Comment (pzhine) [inline packages/databyss-documentation/website/package.json]: Hey, I had to move this file to the root of the package to facilitate the version fix script. Is there a way to keep it there?

File: packages/databyss-ui/components/Editor/EditorBlock.js [hunk 2]
```diff
@@ -1,39 +1,86 @@
...
+        height={editorMarginMenuItemHeight}
+        overflow="visible"
       >
-        {children}
-      </Text>
-    </View>
-  </Grid>
-)
+        {node.text.length < 1 &&
+          !isMobileOs() && (
```
Comment (pzhine) [inline packages/databyss-ui/components/Editor/EditorBlock.js]: Rather than show/hide the menu for mobile, we should be showing/hiding the whole Grid column. In fact, we shouldn't even render it as a Grid on mobile, just a View.

File: packages/databyss-ui/components/Editor/Menu/FormatMenu.js [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+    type: 'location',
+    label: 'loc',
+    variant: 'uiTextNormal',
+    action: a => toggleMark(a),
+  },
+]
+
+const formatActions = isMobileNewLine => [
+  ...(isMobileOs() && isMobileNewLine ? mobileActions : webActions),
+  // {
```
Comment (pzhine) [inline packages/databyss-ui/components/Editor/Menu/FormatMenu.js]: Clean up comments

File: packages/databyss-ui/components/Editor/Menu/FormatMenu.js [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+  },
+  {
+    type: 'LOCATION',
+    label: '+ location',
+    variant: 'uiTextNormal',
+    action: a => startTag(a),
+  },
+]
+
+const webActions = [
```
Comment (pzhine) [inline packages/databyss-ui/components/Editor/Menu/FormatMenu.js]: I'm using the convention `desktop...` for things like this, because `mobile` is web as well. So the three prefixes are `mobile__`, `desktop__` and `native__`

File: packages/databyss-ui/primitives/View/HoverView.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
+import React, { forwardRef } from 'react'
```
Comment (pzhine) [inline packages/databyss-ui/primitives/View/HoverView.js]: Let's rename this `EditorTooltip` and put it in the `components/Editor` directory

File: packages/databyss-ui/primitives/View/HoverView.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+  shadow,
+  layout,
+  compose,
+  border,
+} from 'styled-system'
+import { Grid } from '@databyss-org/ui/primitives'
+import styled from '../styled'
+import IS_NATIVE from '../../lib/isNative'
+
+const variants = variant({
```
Comment (pzhine) [inline packages/databyss-ui/primitives/View/HoverView.js]: Instead of a view variant, let's move those style props into this component, as they are only used in one place rn

File: packages/databyss-ui/primitives/View/HoverView.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+  prop: 'variant',
+  scale: 'hoverVariants',
+  variants: {
+    // need one member to enable theming
+    default: {},
+  },
+})
+
+const Styled = styled(
```
Comment (pzhine) [inline packages/databyss-ui/primitives/View/HoverView.js]: We can just use a `View` primitive here.

File: packages/databyss-ui/theming/views.js [hunk 8]
```diff
@@ -32,7 +32,26 @@ const borderVariants = {
...
 
+const hoverVariants = {
+  formatMenuUI: {
+    // paddingTop: pxUnits(8),
+    paddingLeft: pxUnits(7),
+    // paddingBottom: pxUnits(6),
+    paddingRight: pxUnits(7),
+    backgroundColor: 'background.6',
+    zIndex: 1,
+    top: pxUnits(-10000),
```
Comment (pzhine) [inline packages/databyss-ui/theming/views.js]: Let's try without these...

Comment (pzhine) [reply packages/databyss-ui/theming/views.js]: Instead, set `pointerEvents: 'none'`

File: packages/databyss-ui/components/Editor/state/actions.js.orig [hunk 9]
```diff
@@ -0,0 +1,135 @@
+import {
```
Comment (pzhine) [inline packages/databyss-ui/components/Editor/state/actions.js.orig]: Please remove temp file

File: packages/databyss-ui/components/Editor/slate/reducer.js.orig [hunk 10]
```diff
@@ -0,0 +1,277 @@
+import { Block } from 'slate'
```
Comment (pzhine) [inline packages/databyss-ui/components/Editor/slate/reducer.js.orig]: Please remove temp file

File: packages/databyss-ui/components/Editor/slate/ContentEditable.js.orig [hunk 11]
```diff
@@ -0,0 +1,500 @@
+import React, { useRef, useEffect } from 'react'
```
Comment (pzhine) [inline packages/databyss-ui/components/Editor/slate/ContentEditable.js.orig]: Please remove temp file

File: packages/databyss-ui/components/Editor/state/constants.js.orig [hunk 12]
```diff
@@ -0,0 +1,15 @@
+export const SET_ACTIVE_BLOCK_ID = 'SET_ACTIVE_BLOCK_ID'
```
Comment (pzhine) [inline packages/databyss-ui/components/Editor/state/constants.js.orig]: Please remove temp file

File: packages/databyss-ui/components/Editor/slate/ContentEditable.js [hunk 13]
```diff
@@ -320,28 +347,26 @@ const SlateContentEditable = ({
...
 
-      if (!(selection.isBlurred || selection.isCollapsed)) {
-        if (
-          event.key === 'Backspace' &&
-          !isAtomicNotInSelection(editor.value, editorState)
-        ) {
-          // gets node in the middle of highlight
-          const _node = editor.value.blocks._tail.array[1]
-          deleteBlockByKey(_node.key, editor)
+      if (!isAtomicNotInSelection(editor.value)) {
```
Comment (pzhine) [inline packages/databyss-ui/components/Editor/slate/ContentEditable.js]: Please add comment on which edge case this is handling

File: packages/databyss-ui/components/Editor/slate/ContentEditable.js [hunk 14]
```diff
@@ -320,28 +347,26 @@ const SlateContentEditable = ({
...
+        // if atomic block is highlighted
+        if (editor.value.fragment.nodes.size > 1) {
           return event.preventDefault()
         }
       }
 
+      if (singleBlockBackspaceCheck(editor.value)) {
+        // check selection
```
Comment (pzhine) [inline packages/databyss-ui/components/Editor/slate/ContentEditable.js]: Please add comment on which edge case this is handling

File: packages/databyss-ui/components/Editor/slate/ContentEditable.js [hunk 15]
```diff
@@ -351,22 +376,30 @@ const SlateContentEditable = ({
   }
 
   const onKeyDown = (event, editor, next) => {
-    const { selection } = editor.value
+    const { fragment } = editor.value
+    // check for selection
+    if (hasSelection(editor.value)) {
+      if (event.key === 'Backspace' && !isAtomicNotInSelection(editor.value)) {
```
Comment (pzhine) [inline packages/databyss-ui/components/Editor/slate/ContentEditable.js]: `isAtomicNotInSelection` is confusing
Do you mean `noAtomicInSelection`?

File: packages/databyss-ui/components/Editor/slate/ContentEditable.js [hunk 16]
```diff
@@ -351,22 +376,30 @@ const SlateContentEditable = ({
...
+        return event.preventDefault()
+      }
 
-    // EDGE CASE: prevent block from being deleted when empty block highlighted
-    if (!(selection.isBlurred || selection.isCollapsed)) {
-      if (
-        event.key === 'Backspace' &&
-        !isAtomicNotInSelection(editor.value, editorState)
-      ) {
+      if (singleBlockBackspaceCheck(editor.value)) {
```
Comment (pzhine) [inline packages/databyss-ui/components/Editor/slate/ContentEditable.js]: Is this still handling the one EDGE CASE noted above?

File: packages/databyss-ui/components/Editor/slate/__tests__/bugs.cypress.js [hunk 17]
```diff
@@ -0,0 +1,262 @@
+/** @jsx h */
```
Comment (pzhine) [inline packages/databyss-ui/components/Editor/slate/__tests__/bugs.cypress.js]: Rename to `editor-edge-cases.cypress.js`

File: packages/databyss-ui/components/Editor/slate/__tests__/bugs.cypress.js [hunk 18]
```diff
@@ -0,0 +1,262 @@
...
+            <text />
+          </block>
+        </document>
+      </value>
+    )
+
+    cy.get('@slateDocument').then(matchExpectedJson(expected.document))
+  })
+
+  it('case 3 highlight text bug', () => {
```
Comment (pzhine) [inline packages/databyss-ui/components/Editor/slate/__tests__/bugs.cypress.js]: Paste links to Notion issues above edge case tests, if applicable

File: packages/databyss-ui/components/Editor/EditorTooltip.js [hunk 19]
```diff
@@ -82,39 +72,68 @@ const _activeCss = {
...
+    !selection.focus.isInNode(
+      document.getNode(fragment.nodes.get(fragment.nodes.size - 1).key)
+    )
+  ) {
+    return true
+  }
+  return false
+}
+
+export const getSelectedBlocks = value => {
```
Comment (pzhine) [inline packages/databyss-ui/components/Editor/EditorTooltip.js]: Please add comment on what this method does, as it seems quite complex

File: packages/databyss-ui/components/Editor/EditorTooltip.js [hunk 20]
```diff
@@ -82,39 +72,68 @@ const _activeCss = {
   opacity: 1,
 }
 
-export const isAtomicNotInSelection = (value, editorState) => {
-  const { fragment, selection } = value
+export const isSelectionReversed = value => {
```
Comment (pzhine) [inline packages/databyss-ui/components/Editor/EditorTooltip.js]: What does it mean for a selection to be "reversed"? Please add comment
